src/main/java/essenmakanan/parser/RecipeParser.java

```diff
@@ -38,6 +38,112 @@ public static int getRecipeIndex(RecipeList recipes, String input)
         return index;
     }
 
+    /**
+     * Get a RecipeList with recipes from a recipe ID list
+     *
+     * @param recipeIdList a list of ids of recipes
+     * @param recipes all recipes the user has
+     * @return RecipeList containing of all recipes that correspond to the index in the recipe ID list
+     * @throws EssenOutOfRangeException when the ID in recipe ID list is invalid
+     */
+    public static RecipeList getRecipes(int[] recipeIdList, RecipeList recipes) throws EssenOutOfRangeException {
+        RecipeList allRecipes = new RecipeList();
+        for (int id : recipeIdList) {
+            if (!recipes.recipeExist(id)) {
+                System.out.println("Your recipe Id is wrong");
+                throw new EssenOutOfRangeException();
+            } else {
+                allRecipes.addRecipe(recipes.getRecipe(id));
+            }
```

Comment:
since u have the "if" guard clause that throws and error, line 56 might not need to be in `else`?